packages/cardano-graphql-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/index.ts

```diff
@@ -0,0 +1,167 @@
+    SELECT  
+      tx.hash
+    FROM pool_update pu
+    JOIN tx
+      ON tx.id = pu.registered_tx_id;
+    WHERE pu.hash_id IN $1
+    `,
+      [hashesIds]
+    );
+    return result.rows;
+  }
+  private async queryPoolRelays(updatesIds: number[]) {
+    // missing Relay hostname
+    const result = await this.#db.query(`
+    SELECT
+      relay.update_id as "updateId",
+      relay.ipv4 as ipv4,
+      relay.ipv6 as ipv6,
+      relay.port as port,
+      relay.dns_name as "dnsName"
+    FROM pool_update pr
+    WHERE update_id IN (${updatesIds})
+    `);
+    return result.rows;
+  }
+  private async queryPoolOwners(hashesIds: number[]) {
+    const result = await this.#db.query(
+      `
+    SELECT 
+      stake.view AS address 
+    FROM pool_owner owner
+    JOIN stake_address stake
+      ON stake.id = owner.addr_id
+    WHERE owner.pool_hash_id IN $1
+    `,
+      [hashesIds]
+    );
+    return result.rows;
+  }
+  private async queryPoolRewards(hashesIds: number[], limit?: number) {
+    const params: [number[] | number] = [hashesIds];
+    if (limit) params.push(limit);
+    const result = await this.#db.query(
+      `
+      WITH pool_epoch_reward as (
+        SELECT
+          COUNT(r.amount) as total_rewards,
+          pool_id,
+          r.spendable_epoch as epoch
+        FROM reward r
+        WHERE pool_id IN $1
+        GROUP BY pool_id, spendable_epoch
+        ORDER BY spendable_epoch DESC
+        ${limit ? 'LIMIT = $2' : ''}
+      )
+      SELECT 
+        per.*,
+        epoch.blk_count 
+      FROM pool_epoch_reward per
+      JOIN epoch 
+        ON epoch."no" = per.epoch;
+    `,
+      params
+    );
+    return result.rows;
+  }
+  private async queryLastPoolUpdate(hashId: number) {
+    // Missing PoolParameters.metadata
+    const result = await this.#db.query(
+      `
+      SELECT 
+        pu.hash_id,
+        pu.reward_addr,
+        pu.pledge,
+        pu.fixed_cost,
+        pu.margin,
+        pu.vrf_key_hash,
+        metadata.url, 
+        metadata.hash
+    FROM pool_update pu
+    LEFT JOIN pool_metadata_ref metadata
+      ON metadata.id = pu.meta_id
+    WHERE pu.hash_id = $1
+    `,
+      [hashId]
+    );
+    return result.rows.length > 0 ? result.rows[0] : undefined;
+  }
+  private async queryPoolHashes(options?: StakePoolQueryOptions) {
+    const offset = options?.pagination?.startAt;
+    const limit = options?.pagination?.limit;
+    const result: QueryResult<PoolHashModel> = await this.#db.query(
+      `
+    SELECT 
+      hash.id,
+      hash_raw,
+      view
+    FROM pool_hash hash
+    ${offset ? 'OFFSET = $1' : ''}
+    ${limit ? 'LIMIT = $2' : ''}
+    `,
+      [offset, limit]
+    );
+    return result.rows.length > 0 ? result.rows.map(mapPoolHash) : [];
+  }
+  public async queryStakePools(fragments: string[], options?: StakePoolQueryOptions): Promise<Cardano.StakePool[]> {
+    // TODO: include fragments in the corresponding queries. Which fields can be queried?
```

Comment:
We're going to refactor the provider interface to implement a concrete set of filters in `StakePoolQueryOptions` and remove this GraphQL fragments argument, so you can leave this unimplemented.

Reply:
Interface currently applied